src/main/webapp/scripts/tests/analyze-test.js

```diff
@@ -0,0 +1,67 @@
+const VIDEO_LINK = "https://www.youtube.com/watch?v=testvideoid";
+const SCORE = 0.05;
+
+QUnit.module('Updating DOM after analyzing video');
+QUnit.testStart(() => {
+  document.getElementById('video-url').innerText = VIDEO_LINK;
+})
+
+QUnit.test('DOM updates with sentiment score', function (assert) {
+  const fetchStub = sinon.stub(window, 'fetch');
+  const body = { "score": SCORE, "dataAvailable": true };
+  const response = { json: () => { return body }, status: 200 };
+  fetchStub.returns(Promise.resolve(response));
+
+  const done = assert.async()
+  analyzeVideo();
+  setTimeout(() => {
```

Comment:
using timeout seems like the test would be flaky. I believe Zoe has a solution for this.